bundler/spec/bundler/bundler_spec.rb

```diff
@@ -164,59 +164,31 @@
   end
 
   describe "#which" do
-    let(:executable) { "executable" }
+    it "can detect relative path" do
+      create_file("test_command", "#!/usr/bin/env ruby\n")
 
-    let(:path) do
-      if Gem.win_platform?
-        %w[C:/a C:/b C:/c C:/../d C:/e]
-      else
-        %w[/a /b c ../d /e]
+      result = ""
+      Dir.chdir bundled_app("test_command").parent do
```

Comment:
I think you are using `bundled_app("test_command").parent` rather than `bundled_app` to make it clear that you want to switch to test_command's folder, right?

I'd record the script path and use that. Also, you don't need to explicitly initialize result since it's the result of the block. So, I'd go with something like:

```ruby
script_path = bundled_app("test_command")
create_file(script_path, "#!/usr/bin/env ruby\n")

result = Dir.chdir script_path.dirname do
  Bundler.which("test_command")
end
```

Regardless of this little tweak, tests look indeed much better now :heart:

Reply:
Yes. but the exec defaults to the app directory. I will change to a temp folder to add another test.

Reply:
I did not understand. Tests normally run in the root repository folder. So your approach of switching to `bundled_app` to test relative path support seems fine to me. My suggestion was just to make the code a little cleaner but the test itself looks good to me.

Reply:
ok maybe I mixed it with exec, I mean to do something like this
![image](https://github.com/user-attachments/assets/0457216c-c99a-4ebb-a624-e0c6ed5da265)

Reply:
Aaah I see, that makes sense.